### PR TITLE
[REVIEW] Fix memory leak issue in `drop_duplicate`

### LINKED ISSUE #3269: [BUG] Memory leak on drop_duplicates
**Describe the bug**

Statement `df4 = df3.drop_duplicates([col])` leaks memory, which is problematic for large files that are bigger than GPU RAM. In our case, ~30% of the memory is leaking, so capping us < 50GB files.

**Steps/Code to reproduce bug**
Follow this guide http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

Cannot share underlying data, but overall loop looks like:

1. Compute:
```
def handle_chunk(df, projections, col):
    early_exit = False
    df2 = df[projections]
    stripped = cudf.Series(df[col].str.strip('  	"'))
    replaced = stripped.str.replace('\r','', regex=True)
    df3 = df2.assign(**{col: replaced})
    if early_exit:
        return df3
    df4 = df3.drop_duplicates([col])
    return df4
```

2. Loop:
```
offset = 0
step = 100000
while not done:  
    df = cudf.read_csv(infile, byte_range=[offset,step], names=cols, usecols=cols,
                           dtype=[...])
    df2 = handle_chunk(df, ['some','cols'], 'my_primary_key')
    offset = offset = step
```
    

**Expected behavior**

The value of `early_exit` should not impact termination. This loop should run indefinitely. If an allocation ever fails, gc should free the stale allocations, and ideally, it should happen before then.

Instead, GPU memory steadily increases until explosion. Early existing before the `drop` statement keeps the memory running in a steady state, which is expected.

**Environment overview (please complete the following information)**

Encountered in both of

```
rapidsai/rapidsai-nightly   0.11-cuda10.0-base-ubuntu18.04   ceeab396ef08        14 hours ago        4.02GB
rapidsai/rapidsai           0.10-cuda10.0-base-ubuntu18.04   75ab236316f8        2 weeks ago         3.83GB
```

**Environment details**
Azure nvc6_v2's / ubuntu / nvidia docker

```
Thu Oct 31 21:26:23 2019       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 430.50       Driver Version: 430.50       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P100-PCIE...  Off  | 00008110:00:00.0 Off |                  Off |
| N/A   26C    P0    32W / 250W |   4982MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
```

**Additional context**

https://rapids-goai.slack.com/archives/C5E06F4DC/p1572548162002100

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
It was due to `nvcategory_gather_table`  begin called twice, once in `cudf::gather` and one in `drop_duplicate`.

close #3269

### CODE REVIEW

**Comment by jrhemstad** [comment]
@lmeyerov can you verify that this fixes your memory leak?

**Comment by lmeyerov** [comment]
@jrhemstad will this be in nightly's tmw or..?

File: CHANGELOG.md [hunk 1]
```diff
@@ -209,6 +209,7 @@
 - PR #3439 Fix index-name bug in StringColumn concat
 - PR #3445 Fix ORC Writer default stripe size
 - PR #3459 Fix printing of invalid entries
+- PR #3468 Fix memory leak issue in `drop_duplicate`
```
Comment (harrism) [inline CHANGELOG.md]:
```suggestion
- PR #3468 Fix memory leak issue in `drop_duplicates`
```

**Comment by harrism** [comment]
@lmeyerov Won't be in nightlies until we merge it. But it will be merged as soon as CI passes I think.

For now if you build cudf from source you could test by pulling this branch (see the `command line instructions` link below).

**Comment by lmeyerov** [comment]
Cool, that sounds like it'll land tonight, so will check on the customer setup tmw. Thanks all, the context this is being is used in is prob worth a writeup :)

**Pull request merged**